server/controllers/userController.js

```diff
@@ -17,105 +17,43 @@ export default class UsersController {
     }
     let newUser;
     try {
-      newUser = await user.signUp();
-    } catch (error) {
-      return error.message;
-    }
-
-    const {
-      id, firstname, lastname, email,
-      type, isadmin,
-    } = newUser;
-
-    const token = await Jwt.generateToken({
-      id, email, type, isadmin, firstname, lastname,
-    });
-
-    const response = {
-      token, id, firstname, lastname, email,
-    };
-    return res.status(201).json({
-      status: 201,
-      data: response,
-    });
-  }
-
-  static async createStaff(req, res) {
-    const user = new User(req.body);
-
-    user.password = hashPassword(user.password);
-
-    const userExists = await User.getUserByEmail(user.email);
-    if (userExists) {
-      return res.status(409).json({
-        status: 409,
-        error: 'This email address is already taken.',
-      });
-    }
-
-    let newUser;
-    try {
-      newUser = await user.createStaff();
+      if (req.url.includes('admin')) {
```

Comment:
Hey, great work here @LukasChiama I see you've refactored the tests a little and included the polyfill in the test files. That's great stuff the test script looks a lot cleaner. Thank you for that! Could you please explain what this part of the code is doing?

Reply:
Thank you @tolulope-od.

There were originally three controllers that handled sign up, one for client signup, another for staff signup and the last for admin signup. This involved heavy code repetition as the only difference between these controllers was that they came from different routes. Three of them did the same thing. I removed the controllers and kept just one. From that one controller, I can get the route and do the create the users like the original design.

Following the original design of the API, the route determines what type of user is been signed up. So that line of code checks if the endpoint the request came from is the admin signup endpoint. If it is, the admin model is called and a new admin is created. If it is a staff, then a new staff is created and likewise for clients.

The staff and admin route are protected by appropriate middleware that I added in an earlier bug fix

Reply:
That makes a lot of sense. Great work, I can see you put some thought into that logic. Good stuff man 💯

Reply:
Just as @tolulope-od said, your idea of implementing the feature as you did is cool. However, I don't know if you could find a way of not attaching this endpoint to the request URL that comes from the frontend. What do you think would happen if I want to use this API for another frontend whose UI file system is different from the way the code owner structured hers.

I would suggest you use the `type` and `isAdmin` properties of the user entity. I guess that is universal.

Do you get my explanation? Don't mind my English; I'm not related to the Queen...haha

Reply:
Wow. Thanks a lot @tolulope-od. Thank you for the feedback